File: ruby/darts/darts.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/darts
# Darts exercise
class Darts
  def initialize(point_x, point_y)
    @point_x = point_x
    @point_y = point_y

    @distance = Math.hypot(@point_x, @point_y)
  end

  def score
    return 10 if @distance <= 1.0
    return 5 if @distance <= 5.0
    return 1 if @distance <= 10.0

    0
  end
end

File: lua/darts/darts.lua
local Darts = {}

function Darts.score(x, y)
    local distance = math.sqrt(x ^ 2 + y ^ 2)

    if distance <= 1.0 then
        return 10
    end

    if distance <= 5.0 then
        return 5
    end

    if distance <= 10.0 then
        return 1
    end

    return 0
end

return Darts
